crates/ruff_python_semantic/src/all.rs

```diff
@@ -123,16 +123,16 @@ where
             }) => {
                 // Allow `tuple()`, `list()`, and their generic forms, like `list[int]()`.
                 if arguments.keywords.is_empty() && arguments.args.len() <= 1 {
-                    if let Expr::Name(ast::ExprName { id, .. }) = map_subscript(func) {
+                    if let Expr::Name(ExprName { id, .. }) = map_subscript(func) {
                         let id = id.as_str();
                         if matches!(id, "tuple" | "list") && is_builtin(id) {
```

Comment:
Did you consider making this function a method on the `SemanticModel` struct? It would mean you could easily make use of the `semantic.resolve_builtin_symbol()` method. That would have two advantages:

1. We wouldn't have to pass in the `is_builtin` closure to this function, which feels slightly awkward
2. It's more accurate, as we would recognise `__all__ = builtins.list(["foo", "bar"])` as a valid `__all__` definition, as well as `__all__ = list(["foo", "bar"])`

```suggestion
                    if self
                        .resolve_builtin_symbol(map_subscript(func))
                        .is_some_and(|symbol| matches!(symbol, "tuple" | "list")) {
```

Reply:
I prefer not to make any code changes other than moving the module in this PR

Reply:
Makes sense, I agree that this change by itself is an improvement!